### PR TITLE
Feature/#3 name and bio

### LINKED ISSUE #3: As a personal user I should be able to create a basic profile with name and bio
3. Create a basic profile with name and bio.

### PR DESCRIPTION
**Pull request by revature-ben-m**
Depends on #17

### CODE REVIEW

**Comment by CBlank-dev** [comment]
Looks good

**Review by CBlank-dev — DISMISSED** [review]

**Review by CBlank-dev — DISMISSED** [review]

File: src/test/java/com/rev_connect_api/NameAndBioTest.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+import com.rev_connect_api.models.FieldErrorResponse;
+import com.rev_connect_api.models.PersonalProfile;
+import com.rev_connect_api.models.Role;
+import com.rev_connect_api.models.User;
+import com.rev_connect_api.repositories.ProfileRepository;
+import com.rev_connect_api.repositories.UserRepository;
+import com.rev_connect_api.security.JwtUtil;
+import com.rev_connect_api.utils.UserUtils;
+
+//TODO: Authentication in methods
```
Comment (jurczakn) [inline src/test/java/com/rev_connect_api/NameAndBioTest.java]: If this was done, please remove the comment

File: src/test/java/com/rev_connect_api/NameAndBioTest.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    private User testUser;
+    private String testBio;
+    private String token;
+
+    @BeforeEach
+    public void beforeEach() {
+        serviceLocation = "http://localhost:" + port + "/api/profile";
+        mapper = new ObjectMapper();
+        mapper.findAndRegisterModules();
+        userRepository.deleteAll();
```
Comment (jurczakn) [inline src/test/java/com/rev_connect_api/NameAndBioTest.java]: Can remove dependency for these tests by creating a test sql script to run

File: src/test/java/com/rev_connect_api/NameAndBioTest.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Authorization", "Bearer " + token);
+
+        HttpEntity<PersonalProfile> requestEntity = new HttpEntity<>(newProfile, headers);
+        ResponseEntity<String> response = testRestTemplate.exchange(serviceLocation, HttpMethod.PUT, requestEntity,String.class);
+
+        int statusCode = response.getStatusCode().value();
+        Assertions.assertEquals(200, statusCode, "Expected status code 200. Actual result was " + statusCode);
```
Comment (jurczakn) [inline src/test/java/com/rev_connect_api/NameAndBioTest.java]: Clarify what the status code response should be

File: src/main/java/com/rev_connect_api/controllers/ProfileController.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import com.rev_connect_api.models.FieldErrorResponse;
+import com.rev_connect_api.models.PersonalProfile;
+import com.rev_connect_api.services.ProfileService;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:5173")
+@RequestMapping("/profile")
+public class ProfileController {
+  @Autowired
+  ProfileService profileService;
```
Comment (jurczakn) [inline src/main/java/com/rev_connect_api/controllers/ProfileController.java]: Should we really be using field injection?

Comment (jurczakn) [reply src/main/java/com/rev_connect_api/controllers/ProfileController.java]: Shouldn't this be private?

File: src/main/java/com/rev_connect_api/controllers/ProfileController.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  @GetMapping("/{user_id}")
+  public ResponseEntity<PersonalProfile> retrieveProfile(@PathVariable Long user_id) { 
+    PersonalProfile result;
+    try {
+      result = profileService.retrieveProfile(user_id);
+      return ResponseEntity.status(HttpStatusCode.valueOf(200)).body(result);
+    } catch (InvalidUserException e) {
+      return ResponseEntity.status(HttpStatusCode.valueOf(404)).body(null);
```
Comment (jurczakn) [inline src/main/java/com/rev_connect_api/controllers/ProfileController.java]: comment what the null return value is supposed to mean

File: src/main/java/com/rev_connect_api/services/ProfileService.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  public PersonalProfile retrieveProfile(long user_id) throws InvalidUserException {
+    Optional<PersonalProfile> optionalProfile = profileRepository.findByUser_UserId(user_id);
+    if(!optionalProfile.isPresent()) {
+      throw new InvalidUserException("User " + user_id + " was not found.");
+    }
+    return optionalProfile.get();
+  }
+
+  public PersonalProfile updateProfile(PersonalProfile newProfile) throws InvalidProfileException, InvalidUserException {
+    boolean firstNameEmpty = newProfile.getUser().getFirstName() == "" || newProfile.getUser().getFirstName() == null;
```
Comment (jurczakn) [inline src/main/java/com/rev_connect_api/services/ProfileService.java]: compare using .equals()  "".equals(newProfile.getUser().getFirstName())

**Review by jurczakn — DISMISSED** [review]

File: src/test/java/com/rev_connect_api/NameAndBioTest.java (hunk 3 above)
Comment (revature-ben-m) [reply src/test/java/com/rev_connect_api/NameAndBioTest.java]: 200 seems to be the standard for responding to put requests. I'll make sure the rest of the team is on board with this standard.

File: src/test/java/com/rev_connect_api/NameAndBioTest.java (hunk 1 above)
Comment (revature-ben-m) [reply src/test/java/com/rev_connect_api/NameAndBioTest.java]: Thanks for spotting this! Fixed!

File: src/test/java/com/rev_connect_api/NameAndBioTest.java (hunk 2 above)
Comment (revature-ben-m) [reply src/test/java/com/rev_connect_api/NameAndBioTest.java]: Done!

File: src/main/java/com/rev_connect_api/controllers/ProfileController.java (hunk 4 above)
Comment (revature-ben-m) [reply src/main/java/com/rev_connect_api/controllers/ProfileController.java]: Done and done!

File: src/main/java/com/rev_connect_api/services/ProfileService.java (hunk 6 above)
Comment (revature-ben-m) [reply src/main/java/com/rev_connect_api/services/ProfileService.java]: I haven't seen that syntax before but its very clean. Done!

**Review by themellowpea — APPROVED** [review]
👍🏽

**Review by CBlank-dev — APPROVED** [review]

**Pull request merged**